sympy/geometry/point.py

```diff
@@ -167,20 +224,23 @@ def is_collinear(*args):
 
         return Point.affine_rank(*args) == 1
 
-    def is_scalar_multiple(self, p2):
-        """Returns whether `p1` and `p2` are scalar multiples
-        of each other.
+    def is_scalar_multiple(self, p):
+        """Returns whether each dimension of `p` is a scalar multiple
+        of those of `self`.
         """
-        p1, p2 = Point.pointify(self), Point.pointify(p2)
-
+        s, o = Point.normalize_dimensions(self, Point(p))
         # 2d points happen a lot, so optimize this function call
-        if p1.ambient_dimension == 2:
-            (x1,y1), (x2,y2) = p1.args, p2.args
-            return simplify(x1*y2-x2*y1) == 0
+        if s.ambient_dimension == 2:
+            (x1, y1), (x2, y2) = s.args, o.args
+            rv = (x1*y2 - x2*y1).equals(0)
```

Comment:
I used `equals` here because `simplify` alone isn't powerful enough to resolve some expressions. I found some of those expression intermittently during testing and found that the use of `set` was making it not always happen. So I changed `set` to `uniq` (which preserves order) and used `permutations` in the tests to force such expressions to be encountered.